### ISSUE TITLE
HTTP requests not permitted

### ISSUE THREAD

**Issue by kategray** [openaudible/openaudible#12]
When attempting to load a HTTP URL in the browser, by redirect or otherwise, I get an error about transport security.

<img width="532" alt="screen shot 2018-06-27 at 10 31 39 pm" src="https://user-images.githubusercontent.com/2415200/42009807-f67b0fca-7a59-11e8-8648-151dc4e08056.png">

This error shows on both the new update notification on 1.1.1, as well as attempting to login with 1.1.2.

**Comment by openaudible** [comment]
What region are you trying to connect to? (US, Canada, etc.) Did you set that in the preferences?
Can you open the OpenAudible browser (CMD-B) and navigate to your library and view your books? 
Can you copy and paste the URL you're seeing when you view your books?
Also, restart the app, open the console window, and reproduce the error. Then copy and paste the console (removing any PII) and send by pastebin or simply post here..

**Comment by kategray** [comment]
It appears to be an issue with OS X.  I was able to resolve it by adding the following to Info.plist

`<key>NSAppTransportSecurity</key>
<dict>
    <key>NSExceptionDomains</key>
    <dict>
        <key>audible.com</key>
        <dict>
            <key>NSIncludesSubdomains</key>
            <true/>
            <key>NSExceptionAllowsInsecureHTTPLoads</key>
            <true/>
            <key>NSExceptionRequiresForwardSecrecy</key>
            <true/>
            <key>NSExceptionMinimumTLSVersion</key>
            <string>TLSv1.2</string>
            <key>NSThirdPartyExceptionAllowsInsecureHTTPLoads</key>
            <false/>
            <key>NSThirdPartyExceptionRequiresForwardSecrecy</key>
            <true/>
            <key>NSThirdPartyExceptionMinimumTLSVersion</key>
            <string>TLSv1.2</string>
            <key>NSRequiresCertificateTransparency</key>
            <false/>
        </dict>
    </dict>
</dict>
`

It's a [Cocoa Thing](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html) apparently.

> On Apple platforms, a networking security feature called App Transport Security (ATS) is available to apps and app extensions, and is enabled by default. It improves privacy and data integrity by ensuring your app’s network connections employ only industry-standard protocols and ciphers without known weaknesses. This helps instill user trust that your app does not accidentally leak transmitted data to malicious parties.

> By configuring this key’s value in your app’s Info.plist file, you can customize the security of your network connections in a variety of ways. You can:

> - Allow insecure communication with particular servers
> - Allow insecure loads for web views or for media, while maintaining ATS protections elsewhere in your app
> - Enable new security features such as Certificate Transparency

Basically, the problem seems to be that unless an option is set, Apple isn't letting apps use HTTP.  Audible still has some redirects that use HTTP, and they fail when that happens.

**Comment by openaudible** [comment]
Thanks for the detailed report! I added your plist fragment to the mac application. It runs, but will need you to test it. https://github.com/openaudible/openaudible/releases/latest
Appreciate the help.
